modules/auxiliary/admin/citrix/citrix_netscaler_config_decrypt.rb

```diff
@@ -0,0 +1,259 @@
+      end
+    end
+    parse_ns_config
+  end
+
+  def build_ns_kek
+    unless File.size(ns_kek_f1) == 256 && File.size(ns_kek_f2) == 256
+      print_error('KEK files must be 256 bytes in size')
+      return false
+    end
+    f1_hex = File.read(ns_kek_f1)
+    f2_hex = File.read(ns_kek_f2)
+    f1_hex&.encode('UTF-8', 'binary')
+    f2_hex&.encode('UTF-8', 'binary')
+    unless f1_hex.match?(/^[0-9a-f]+$/i) && f2_hex.match?(/^[0-9a-f]+$/i)
+      print_error('NS KEK files must be hexadecimal format')
+      return false
+    end
+    f1_key = f1_hex[66..130].scan(/../).map(&:hex).pack('C*')
+    f2_key = f2_hex[70..134].scan(/../).map(&:hex).pack('C*')
+    f1_key_hex = f1_key.unpack('H*').first
+    f2_key_hex = f2_key.unpack('H*').first
+    print_good('NS KEK F1')
+    print_good("\t HEX: #{f1_key_hex}")
+    print_good('NS KEK F2')
+    print_good("\t HEX: #{f2_key_hex}")
+    @ns_kek_key = OpenSSL::HMAC.hexdigest('SHA256', f2_key, f1_key).scan(/../).map(&:hex).pack('C*')
+    @ns_kek_key_hex = @ns_kek_key.unpack('H*').first
+    print_good('Assembled NS KEK AES key')
+    print_good("\t HEX: #{@ns_kek_key_hex}\n")
+    true
+  rescue Encoding::UndefinedConversionError
+    print_error('Invalid NS KEK files provided: invalid UTF-8 data')
+    return false
+  rescue StandardError => e
+    print_error("#{__method__}: #{e.message}")
+    return false
+  end
+
+  def parse_ns_config
+    ns_config_data = File.binread(ns_conf)
+    ns_secret.each do |secret|
+      element = secret[0]
+      secret[1].each do |keyword|
+        lines = ns_config_data.to_enum(:scan, /^#{keyword}.*/).map { Regexp.last_match }
+        lines.each do |line|
+          is_kek = false
+          config_entry = line.to_s
+          ciphertext = config_entry.to_enum(:scan, /#?([\da-f]{2})([\da-f]{2})([\da-f]{2})(\w+)/).map { Regexp.last_match }
+          unless ciphertext.first
+            ciphertext = config_entry.to_enum(:scan, /(-passcrypt.*(\s*))/).map { Regexp.last_match }
+            next unless ciphertext.first
+          end
+          enc_type = config_entry.match(/encryptmethod (\w+)/).to_s.split(' ')[1].to_s
+          if config_entry.match?(/-kek/)
+            is_kek = true
+          end
+          print_status("Config line:\n#{config_entry}")
+          if is_kek && !@ns_kek_key
+            print_warning('Entry was encrypted with KEK but no KEK fragement files provided, decryption will not be possible')
+            next
+          end
+          username = parse_username_from_config(config_entry)
+          ciphertext.each do |encrypted|
+            encrypted_entry = encrypted.to_s
+            if encrypted_entry =~ /^[0-9a-f]+$/i
+              ciphertext_bytes = encrypted_entry.scan(/../).map(&:hex).pack('C*')
+            else
+              ciphertext_b64 = encrypted_entry.split(' ')[1].delete('"')
+              ciphertext_bytes = Base64.strict_decode64(ciphertext_b64)
+              # Still haven't grokked how -passcrypt works
```

Comment:
```suggestion
              # TODO Implement -passcrypt functionality
              # ciphertext_bytes = Base64.strict_decode64(ciphertext_b64)
```
`ciphertext_bytes` doesn't get used in this context, refrain from initializing it. Also it might be worth while to add a TODO here for -passcrypt functionality?

Reply:
Works for me - I was hoping to have had time to figure out -passcrypt by now, but it has been too crazy.